File: Concesionario/src/gestorAplicacion/activos/InventarioAuto.java
package gestorAplicacion.activos;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class InventarioAuto implements Serializable{
	private static final long serialVersionUID = 1L;
	static ArrayList<Auto> autos = new ArrayList<Auto>();

	public void agregarAuto(Auto auto) {
		autos.add(auto);
	}

	public void eliminarAuto(Auto auto) {
		autos.remove(auto);
	}

	public static ArrayList<Auto> getAutos() {
		return autos;
	}

	public static void setAutos(ArrayList<Auto> autos) {
		InventarioAuto.autos = autos;
	}

	public static String autosDisponibles() {
		int cont = 1;
		String resultado = "Autos Disponibles:\n";
		for (Auto auto : getAutosDisponibles()) {
			resultado += cont + ". " + auto.getModelo() + " " +  auto.getMarca() + " " + auto.getPrecio() + " " + auto.getColor() + "\n";
			cont++;
		}
		return resultado;
	}

	public static ArrayList<Auto> getAutosDisponibles() {
		ArrayList<Auto> disponibles = new ArrayList<Auto>();
		for (Auto auto : autos) {
			if (auto.isDisponible()) {
				disponibles.add(auto);
			}
		}
		return disponibles;
	}
	
	public static ArrayList<Auto> AutosVendidos(ArrayList<Auto> autosinic) {
		ArrayList<Auto> vendidos = new ArrayList<>();
		for (Auto auto : autosinic) {
			if (auto.isDisponible()) {
			} else {
				vendidos.add(auto);
			}
		}
		return vendidos;
	}
	
	public static Auto getAutoporModelo(String modelo) {
		Auto finder = null;
		for (Auto auto : getAutosDisponibles()) {
			if(modelo==auto.getModelo()) {
				finder = auto;
			}else {
				return finder;
			}
		}return finder;
	}
	
	public static ArrayList<Auto> getAutosporPrecio() {
		ArrayList<Auto> autosPrecio = new ArrayList<Auto>();
		for (Auto auto: InventarioAuto.getAutosDisponibles()) {
			autosPrecio.add(auto);
		}
		Collections.sort(autosPrecio, new Sortbyroll());
		return autosPrecio;
	}
	
	public static ArrayList<Auto> getAutosporModelo(String modelo) {
		ArrayList<Auto> modelosInteres = new ArrayList<Auto>();
		for (Auto auto: getAutosDisponibles()) {
			if(modelo==auto.getModelo()) {
				modelosInteres.add(auto);
			}
		}if (modelosInteres.size()==0) {
			System.out.println("No existen carros de este modelo disponibles en este momento.");
		}return modelosInteres;
	}
	
	public static byte readByte() {
	    Scanner scanner = new Scanner(System.in);
	    byte num = scanner.nextByte();
	    return num;
	}

	



}

File: PYTHON/Concesionario/gestorAplicacion/Activos/InventarioAuto.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)
from Activos.auto import Auto
class InventarioAuto:
    autos = []

    def agregarAuto(self, auto):
        self.autos.append(auto)

    def eliminarAuto(self, auto):
        self.autos.remove(auto)

    @classmethod
    def get_autos(cls):
        return cls.autos

    @classmethod
    def set_autos(cls,aut):
        cls.autos = aut

    @staticmethod
    def autos_disponibles():
        cont = 1
        resultado = "Autos Disponibles:\n"
        for auto in InventarioAuto.get_autos_disponibles():
            resultado += f"{cont}. {auto.modelo} {auto.marca} {auto.precio} {auto.color}\n"
            cont += 1
        return resultado

    @staticmethod
    def get_autos_disponibles():
        disponibles = []
        for auto in Auto.autos:
            if auto.disponible:
                disponibles.append(auto)
        return disponibles

    @staticmethod
    def autos_vendidos(autosinic):
        vendidos = []
        for auto in autosinic:
            if auto.disponible:
                pass
            else:
                vendidos.append(auto)
        return vendidos

    @staticmethod
    def get_autoporModelo(modelo):
        finder = None
        for auto in InventarioAuto.get_autos_disponibles():
            if modelo == auto.modelo:
                finder = auto
            else:
                return finder
        return finder

    @staticmethod
    def get_autosporPrecio(cliente):
        autosPrecio = []
        for auto in InventarioAuto.get_autos_disponibles():
            if cliente.presupuesto >= auto.precio:
                autosPrecio.append(auto)
        autosPrecio.sort(key=lambda x: x.precio)
        return autosPrecio

    @staticmethod
    def get_autosporModelo(modelo):
        modelosInteres = []
        for auto in InventarioAuto.get_autos_disponibles():
            if modelo == auto.marca:
                modelosInteres.append(auto)
        if len(modelosInteres) == 0:
            print("No existen carros de este modelo disponibles en este momento.")
        return modelosInteres

    @staticmethod
    def readByte():
        num = int(input())
        return num